controllers/utils/dns.go

```diff
@@ -64,8 +69,42 @@ func Dig(fqdn string, edgeDNSServers ...DNSServer) (ips []string, err error) {
 	if err != nil {
 		return nil, fmt.Errorf("dig error: %s", err)
 	}
+	aRecords := make([]*dns.A, 0)
+	cnameRecords := make([]*dns.CNAME, 0)
 	for _, a := range ack.Answer {
-		ips = append(ips, a.(*dns.A).A.String())
+		switch v := a.(type) {
+		case *dns.A:
+			ips = append(ips, v.A.String())
+			aRecords = append(aRecords, v)
+		case *dns.CNAME:
+			cnameRecords = append(cnameRecords, v)
+		}
+	}
+	aResolved := func(c *dns.CNAME) bool {
+		for _, a := range aRecords {
+			if c.Target == a.Hdr.Name {
+				return true
+			}
+		}
+		return false
+	}
+	cResolved := func(c *dns.CNAME) bool {
+		for _, cname := range cnameRecords {
+			if c.Target == cname.Hdr.Name {
```

Comment:
re-reading this a few hours later, I see that I can actually just use the same anon function for both aResolved and cResolved.  I'll fixup in a bit.